components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/utils/ImportUtils.java

```diff
@@ -0,0 +1,287 @@
+            if (log.isDebugEnabled()) {
+                log.debug("Found application definition file " + pathToJsonFile);
+            }
+            jsonContent = FileUtils.readFileToString(new File(pathToJsonFile));
+        }
+        return jsonContent;
+    }
+
+    /**
+     * Check whether a provided userId corresponds to a valid consumer of the store and subscribe if valid
+     *
+     * @param userId      Username of the Owner
+     * @param groupId     The groupId to which the target subscriber belongs to
+     * @param apiConsumer API Consumer
+     * @throws APIManagementException if an error occurs while checking the validity of user
+     */
+    public static void validateOwner(String userId, String groupId, APIConsumer apiConsumer)
+            throws APIManagementException {
+        Subscriber subscriber = apiConsumer.getSubscriber(userId);
+        try {
+            if (subscriber == null && !APIUtil.isPermissionCheckDisabled()) {
+                APIUtil.checkPermission(userId, APIConstants.Permissions.API_SUBSCRIBE);
+                apiConsumer.addSubscriber(userId, groupId);
+            }
+        } catch (APIManagementException e) {
+            throw new APIManagementException("Provided Application Owner is Invalid", e);
+        }
+    }
+
+    /**
+     * This extracts information for creating an APIKey from an OAuthApplication
+     *
+     * @param applicationKeyDto Application Key DTO
+     * @return An APIKey containing keys from OAuthApplication
+     */
+    public static APIKey getAPIKeyFromApplicationKeyDTO(ApplicationKeyDTO applicationKeyDto) {
+        APIKey apiKey = new APIKey();
+        apiKey.setType(String.valueOf(applicationKeyDto.getKeyType()));
+        apiKey.setConsumerKey(applicationKeyDto.getConsumerKey());
+        apiKey.setConsumerSecret(new String(Base64.decodeBase64(applicationKeyDto.getConsumerSecret())));
+        apiKey.setKeyManager(applicationKeyDto.getKeyManager());
+        apiKey.setGrantTypes(StringUtils.join(applicationKeyDto.getSupportedGrantTypes(), ", "));
+        if (apiKey.getGrantTypes() != null && apiKey.getGrantTypes().contains(GRANT_TYPE_IMPLICIT) && apiKey
+                .getGrantTypes().contains(GRANT_TYPE_CODE)) {
+            apiKey.setCallbackUrl(applicationKeyDto.getCallbackUrl());
+        }
+        apiKey.setValidityPeriod(applicationKeyDto.getToken().getValidityTime());
+        apiKey.setTokenScope(DEFAULT_TOKEN_SCOPE);
+        return apiKey;
+    }
+
+    /**
+     * Import and add subscriptions of a particular application for the available APIs and API products
+     *
+     * @param subscribedAPIs Subscribed APIs
+     * @param userId         Username of the subscriber
+     * @param appId          Application Id
+     * @param apiConsumer    API Consumer
+     * @return a list of APIIdentifiers of the skipped subscriptions
+     * @throws APIManagementException if an error occurs while importing and adding subscriptions
```

Comment:
Missing update param value and an exception

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/commit/eb9b7181c0f6cb5b9624602d91d28a7479bf551b